src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandWrapperJsonConverter.cs

```diff
@@ -0,0 +1,308 @@
+//-----------------------------------------------------------------------------
+// <copyright file="SelectExpandWrapperJsonConverter.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData.Query.Expressions
+{
+    /// <summary>
+    /// Represents a custom <see cref="JsonConverter"/> to serialize <see cref="SelectExpandWrapper{TElement}"/> instances to JSON.
+    /// </summary>
+    internal class SelectExpandWrapperJsonConverter : JsonConverterFactory
+    {
+        public static readonly Func<IEdmModel, IEdmStructuredType, IPropertyMapper> _mapperProvider =
```

Comment:
Where is the `IPropertyMapper` defined? And does it only map the property name or also the value?

Reply:
`IPropertyMapper` is an interface defined in `Microsoft.AspNet.OData.Query` namespace. It maps property names only. It's relevant where the property name to be written on the wire is different from the property name on the type. For example:
```csharp
public class TestEntity
{
    [JsonProperty("Prop")]
    public string Property { get; set; }
}
```